var/spack/repos/builtin/packages/ncl/package.py

```diff
@@ -37,15 +37,15 @@ class Ncl(Package):
 
     homepage = "https://www.ncl.ucar.edu"
 
-    version('6.4.0', 'a981848ddcaf1c263279648265f24766',
-            url='https://www.earthsystemgrid.org/download/fileDownload.html?logicalFileId=86b9bec2-fa01-11e6-a976-00c0f03d5b7c',
+    version('6.4.0', 'd891452cda7bb25afad9b6c876c73986',
+            url='https://codeload.github.com/NCAR/ncl/tar.gz/6.4.0',
```

Comment:
The first location redirects you to the one that I have added

Reply:
The only problem with that URL is that `spack versions` and `spack checksum` won't pick it up. I would switch to the first location and make it a package-level URL instead of a version-specific URL.